internal/decoder/string.go

```diff
@@ -247,14 +247,15 @@ func stringBytes(s *Stream) ([]byte, error) {
 				}
 				goto ERROR
 			}
-			r, _ := utf8.DecodeRune(s.buf[cursor:])
-			b := []byte(string(r))
+			r, size := utf8.DecodeRune(s.buf[cursor:])
 			if r == utf8.RuneError {
-				s.buf = append(append(append([]byte{}, s.buf[:cursor]...), b...), s.buf[cursor+1:]...)
+				s.buf = append(append(append([]byte{}, s.buf[:cursor]...), runeErrBytes...), s.buf[cursor+1:]...)
+				cursor += runeErrBytesLen
+				s.length += runeErrBytesLen
 				_, _, p = s.stat()
+			} else {
+				cursor += int64(size)
```

Comment:
Need to `s.length += int64(size)` ?

Reply:
I think that does not need.
That was the root cause of this issue.